src/org/labkey/targetedms/query/SamplePivotCustomizer.java

```diff
@@ -0,0 +1,89 @@
+package org.labkey.targetedms.query;
+
+import org.apache.commons.collections4.MultiValuedMap;
+import org.labkey.api.collections.CaseInsensitiveHashSet;
+import org.labkey.api.data.CompareType;
+import org.labkey.api.data.SimpleFilter;
+import org.labkey.api.data.TableCustomizer;
+import org.labkey.api.data.TableInfo;
+import org.labkey.api.data.TableSelector;
+import org.labkey.api.query.FieldKey;
+import org.labkey.api.view.HttpView;
+import org.labkey.api.view.ViewContext;
+import org.labkey.targetedms.TargetedMSManager;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Customizes the set of columns available on a pivot query that operates on samples, hiding all of the pivot values
+ * that aren't part of the run that's being filtered on. This lets you view a single document's worth of data
+ * without seeing empty columns for all of the other samples in the same container.
+ */
+public class SamplePivotCustomizer implements TableCustomizer
```

Comment:
Is this supposed to be referenced from some query.xml?

Reply:
Yes, it's a custom report on a customer server